### ISSUE TITLE
SendMessage : Match not found

### ISSUE THREAD

**Issue by Mashallow** [alkawryk/tinderjs#13]
Hi, I retreived the userId (_id) from the getHistory matches, but then, when I try to send a message, it returns : null { status: '500', error: 'Match not found' }
Any idea why?

**Comment by austin1237** [comment]
same issue here

**Comment by austin1237** [comment]
You actually need to send the id of the match instead of the user. The documentation for using the userid  from the pluck is wrong.

**Comment by tnasc1** [comment]
@austin1237 : how do you make the sendMessage function incorporate the matchid instead of the userid?

**Comment by kuvic16** [comment]
@tnastine : you have to pass match_id instead of the user_id. match_id can be found after like response.

**Comment by tnasc1** [comment]
@kuvic16 do you have an example of it? sorry, i've been having trouble. I'm using the example code below:

client.getRecommendations(10, function(error, data){
      _.chain(data.results)
        .pluck('_id')
        .each(function(id) {
          client.like(id, function(error, data) {
            if (data && data.matched) {
              client.sendMessage(id, "message");

**Comment by wcalhoun** [comment]
Same as the above user, can you provide an example?

**Comment by kuvic16** [comment]
@tnastine @wcalhoun : 
See the "data" response from "client.like" by console. You will see match id.
client.like(id, function(error, data) {
   console.log(data);
   client.sendMessage(data._id, "message", function(error, msg){
      console.log(msg);
   });
});

**Comment by tnasc1** [comment]
@kuvic16 

console is spitting back "TypeError: Cannot read property '_id' of null"

**Comment by tnasc1** [comment]
console shows match data when following code runs without the sendMessage function:

client.getRecommendations(recs_size, function(error, data){
      _.chain(data.results)
        .pluck('_id')
        .each(function(id) {
          client.like(id, function(error, data) {  
   console.log(data);
              });
          });

only when I include the sendMessage does it display null error --

**Comment by tnasc1** [comment]
sorry for spamming -- just working this out as I go along.

I think for it to work it needs to message each matched recommendation -- right now its only doing the first one which never seems to be a match according to the data displayed in console. How can I make it message each match it made?

Thanks, I really appreciate it

**Comment by tnasc1** [comment]
GOT IT!!!

thank you @kuvic16 for getting me started

@wcalhoun here's the code I'm using that is working!

client.getRecommendations(1, function(error, data){
      _.chain(data.results)
        .pluck('_id')
        .each(function(id) {
          client.like(id, function(error, data) {  
            console.log(data);
                if(data && data.match){
                    client.sendMessage(data.match._id, "hi", function(error, msg){
                        console.log(msg);

**Comment by wcalhoun** [comment]
Hey @kuvic16 and @tnastine thanks for the help but I was wondering if you could help me decipher the output I am getting from the corrected code:

![capture](https://cloud.githubusercontent.com/assets/10743988/6261293/571987a4-b7bb-11e4-825e-2f83cc498fb6.PNG)

Did it execute correctly?

**Comment by wcalhoun** [comment]
Actually here is the full image not cutoff at the top, sorry for the spam!
![capture](https://cloud.githubusercontent.com/assets/10743988/6261324/9b713b72-b7bb-11e4-9501-c862117c312f.PNG)

**Comment by tnasc1** [comment]
@wcalhoun 

yes, that is the correct output. it is displaying all of the data pulled after liking the given recommendations.

wherever it says "null" or "match: false" is a recommendation that you have not matched with, anywhere with "match _id" is a match -- the client.sendMessage(data.match._id, "message", function(error, msg) function then pulls that match id and sends the message to it.

**Comment by wcalhoun** [comment]
Thank you for the help in understanding the output. 
What happens to the all the matches that happen to my bot profile while this script is not being run?
Just it save them until the next time I run it or just never send a message to those matches?

From William Calhoun

**Comment by kuvic16** [comment]
@tnastine :+1:  You are welcome.
@wcalhoun : By user history. where you can get all matched profile before you did. Just grab that and do whatever you want. Hope it will help you.
